File: leetcode/217. Contains Duplicate/solution.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    bool hasDuplicate(vector<int> &nums)
    {
        set<int> uniqueElements;
        for (int num : nums)
        {
            uniqueElements.insert(num);
        }
        return uniqueElements.size() != nums.size();
    }
};

int main()
{
    Solution sol;
    int n;
    cin >> n;
    vector<int> nums;
    for (int i = 0; i < n; i++)
    {
        int x;
        cin >> x;
        nums.push_back(x);
    }
    cout << sol.hasDuplicate(nums) << endl;
    return 0;
}

File: leetcode/217. Contains Duplicate/solution.java
import java.util.Set;
import java.util.HashSet;
import java.util.Scanner;
class Solution {
    public boolean hasDuplicate(int[] nums) {
    Set<Integer> uniq = new HashSet<>();
        for (int num : nums) {
         uniq.add(num);
        }       
        return uniq.size() != nums.length;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int n;
        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }
        boolean result = solution.hasDuplicate(nums);
        System.out.println(result);
        scanner.close();
    }
}